tests/PronunciationAssessmentTests.ts

```diff
@@ -0,0 +1,172 @@
+    expect(r).not.toBeUndefined();
+
+    return r;
+};
+
+const BuildSpeechConfig: () => sdk.SpeechConfig = (): sdk.SpeechConfig => {
+
+    let s: sdk.SpeechConfig;
+    if (undefined === Settings.SpeechEndpoint) {
+        s = sdk.SpeechConfig.fromSubscription(Settings.SpeechSubscriptionKey, Settings.SpeechRegion);
+    } else {
+        s = sdk.SpeechConfig.fromEndpoint(new URL(Settings.SpeechEndpoint), Settings.SpeechSubscriptionKey);
+    }
+
+    if (undefined !== Settings.proxyServer) {
+        s.setProxy(Settings.proxyServer, Settings.proxyPort);
+    }
+
+    expect(s).not.toBeUndefined();
+    return s;
+};
+
+test("testPronunciationAssessmentConfig::normal", (done: jest.DoneCallback) => {
+    // tslint:disable-next-line:no-console
+    console.info("Name: testPronunciationAssessmentConfig:::normal");
+    let pronConfig: sdk.PronunciationAssessmentConfig = new sdk.PronunciationAssessmentConfig("reference");
+    let j = JSON.parse(pronConfig.toJSON());
+    expect(j.referenceText === "reference");
+    expect(j.gradingSystem === "FivePoint");
+    expect(j.granularity === "Phoneme");
+    expect(j.dimension === "Comprehensive");
+    expect(j.scenarioId).toBeUndefined();
+
+    pronConfig = new sdk.PronunciationAssessmentConfig("reference",
+        PronunciationAssessmentGradingSystem.HundredMark,
+        PronunciationAssessmentGranularity.Word, true);
+    pronConfig.referenceText = "new reference";
+    j = JSON.parse(pronConfig.toJSON());
+    expect(j.referenceText === "new reference");
+    expect(j.gradingSystem === "HundredMark");
+    expect(j.granularity === "Word");
+    expect(j.dimension === "Comprehensive");
+    expect(j.enableMiscue === true);
+    done();
+});
+
+test("testPronunciationAssessmentConfig::fromJson", (done: jest.DoneCallback) => {
+    // tslint:disable-next-line:no-console
+    console.info("Name: testPronunciationAssessmentConfig::fromJson");
+    const jsonString = `{"dimension": "Comprehensive", "enableMiscue": false, "key": "value"}`;
+    const pronConfig = sdk.PronunciationAssessmentConfig.fromJSON(jsonString);
+    expect(JSON.parse(pronConfig.toJSON()) === JSON.parse(jsonString));
+    done();
+});
+
+describe.each([true, false])("Service based tests", (forceNodeWebSocket: boolean) => {
+
+    beforeAll(() => {
+        WebsocketMessageAdapter.forceNpmWebSocket = forceNodeWebSocket;
+    });
+
+    afterAll(() => {
+        WebsocketMessageAdapter.forceNpmWebSocket = false;
+    });
+
+    test("test Pronunciation Assessment", (done: jest.DoneCallback) => {
+        // tslint:disable-next-line:no-console
+        console.info("Name: test Pronunciation Assessment");
+        const s: sdk.SpeechConfig = BuildSpeechConfig();
+        objsToClose.push(s);
+
+        const r: sdk.SpeechRecognizer = BuildRecognizerFromWaveFile(s, Settings.WaveFile);
+        objsToClose.push(r);
+
+        const p: sdk.PronunciationAssessmentConfig = new sdk.PronunciationAssessmentConfig("");
```

Comment:
Can you put some reference text here? empty reference text actually doesn't trigger our score AM on service side.

Reply:
as we talked offline. current test region doesn't support pronunciation assessment. will update the test case after pronunciation assessment is enabled on that region.